### PR TITLE
Watch Later

### PR DESCRIPTION
**Pull request by Jay-Gadhiya**
Functionality added:
- on the card, by clicking on watch later, added to wathlater page
- by click again will remove from watch later
- responsive watchlater page
- modal is added for option menu
- click on watchlater card in watchlater page will redirect to single video page

### CODE REVIEW

File: src/Components/videoCard/videoCard.jsx [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+    const [menu, setMenu] = useState(false);
+    const { authState } = useAuth();
+    const { dataStoreState, dataStoreDispatch } = useData();
+    const navigate = useNavigate();
+
+    const isWatched = dataStoreState.watchLater.find(item => item._id === video._id);
+
+    const menuBoxRef = useClickOutside(() => {
+        setMenu(false);
+    });
```
Comment (gopalbbharadva) [inline src/Components/videoCard/videoCard.jsx]: It can be one liner.

File: src/Pages/Video-player/videoPlayer.jsx [hunk 2]
```diff
@@ -38,7 +38,6 @@ export const VideoPlayer = () => {
                                 url={`https://www.youtube.com/watch?v=${videoId}`}
```
Comment (gopalbbharadva) [inline src/Pages/Video-player/videoPlayer.jsx:41]: Suggestion: you can make utility function.

File: src/Pages/watch-later/watchLater.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        <>
+            <Navbar />
+
+            <div className="aside-main-flex">
+                <Aside />
+
+                <div className="horizontal-main-container">
+                    <div className="like-img-and-info">
+                        <img className="horizontal-hero-img" src={watchLaterImg} alt="like" />
+                        <p className="liked-vid-heading">Liked videos</p>
```
Comment (gopalbbharadva) [inline src/Pages/watch-later/watchLater.jsx]: Typo: Change the typo to watch later videos.

**Review by gopalbbharadva — APPROVED** [review]
functionality working fine.

P.S: add watch later functionality in single video page also.

**Comment by Jay-Gadhiya** [comment]
> 
Done

**Comment by Jay-Gadhiya** [comment]
> functionality working fine.
> 
> P.S: add watch later functionality in single video page also.

Done

**Review by AtulPant2704 — APPROVED** [review]

**Pull request merged**